src/main/groovy/org/gradle/android/workarounds/room/JavaCompileWorkaround.groovy

```diff
@@ -36,38 +36,41 @@ class JavaCompileWorkaround extends AnnotationProcessorWorkaround<JavaCompilerRo
             @Override
             Closure<?> getOldVariantConfiguration() {
                 return { variant ->
-                    // Make sure that the annotation processor argument has not been explicitly configured in the Android
-                    // configuration (i.e. we only want this configured through the room extension)
-                    Map<String, String> arguments = variant.javaCompileOptions.annotationProcessorOptions.arguments
-                    errorIfRoomSchemaAnnotationArgumentSet(arguments.keySet())
-
-                    def variantSpecificSchemaDir = project.objects.directoryProperty()
-                    variantSpecificSchemaDir.set(androidVariantProvider.getVariantSpecificSchemaDir(project, "compile${variant.name.capitalize()}JavaWithJavac"))
-
-                    // Add a command line argument provider to the compile task argument providers
-                    variant.javaCompileProvider.configure { JavaCompile task ->
-                        task.options.compilerArgumentProviders.add(
-                            new JavaCompilerRoomSchemaLocationArgumentProvider(roomExtension.schemaLocationDir, variantSpecificSchemaDir)
-                        )
+                    if (!kspIsAppliedWithRoom()) {
```

Comment:
I know you wrote in the PR description that if room is in both it will only be applied to one. Might be worth a short comment here as a reminder for the future.

Reply:
done https://github.com/gradle/android-cache-fix-gradle-plugin/pull/429/commits/1d0efd0a134c6680a44486d5ae9c3f7ca0cacb2e